test/e2e/apimachinery/aggregator.go

```diff
@@ -103,6 +109,121 @@ var _ = SIGDescribe("Aggregator", func() {
 		TestSampleAPIServer(f, aggrclient, imageutils.GetE2EImage(imageutils.APIServer))
 	})
 
+	ginkgo.It("should manage the lifecycle of an APIService", func() {
+
+		ns := f.Namespace.Name
+		framework.Logf("ns: %v", ns)
+
+		subDomain := "e2e-" + utilrand.String(5)
+		apiServiceGroup := subDomain + ".example.com"
+		label := map[string]string{"e2e": subDomain}
+		labelSelector := labels.SelectorFromSet(label).String()
+
+		apiServiceName := "v1alpha1." + apiServiceGroup
+		apiServiceClient := aggrclient.ApiregistrationV1().APIServices()
+		certCtx := setupServerCert(ns, "e2e-api")
+
+		ginkgo.By(fmt.Sprintf("Create APIService %s", apiServiceName))
+		_, err := apiServiceClient.Create(context.TODO(), &apiregistrationv1.APIService{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   apiServiceName,
+				Labels: label,
+			},
+			Spec: apiregistrationv1.APIServiceSpec{
+				Service: &apiregistrationv1.ServiceReference{
+					Namespace: ns,
+					Name:      "e2e-api",
+					Port:      pointer.Int32Ptr(aggregatorServicePort),
+				},
+				Group:                apiServiceGroup,
+				Version:              "v1alpha1",
+				CABundle:             certCtx.signingCert,
+				GroupPriorityMinimum: 2000,
+				VersionPriority:      200,
+			},
+		}, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "creating apiService %s, namespace %s", apiServiceName, ns)
+
+		_, err = apiServiceClient.List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+		framework.ExpectNoError(err, "failed to list API Services")
+
+		ginkgo.By("Confirm that the generated APIService has been created")
+		err = wait.PollImmediate(apiServiceRetryPeriod, apiServiceRetryTimeout, checkApiServiceListQuantity(aggrclient, labelSelector, 1))
+		framework.ExpectNoError(err, "failed to count the required APIServices")
+
+		ginkgo.By(fmt.Sprintf("Update status for APIService %s", apiServiceName))
+		var statusToUpdate, updatedStatus *apiregistrationv1.APIService
+
+		updatedStatusConditions := apiregistrationv1.APIServiceCondition{
+			Type:    "StatusUpdate",
+			Status:  "True",
+			Reason:  "E2E",
+			Message: "Set from e2e test",
+		}
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Creating an api service can potentially cause alerts and other failure detection to fire in a conformant distribution, since apiservices that don't resolve are significant.  In a sense, this test is creating disruption.

The difference between this test and the others in this file is that those create a service, which means the apiservice is "valid" (has a backend that responds to requests, and the controller for apiservice can find).  In this test, we aren't doing that, which may result in the apiserver reporting "unhealthy".

During the execution of this test, are *other* conditions set on the created apiservice object?  Can you print that out and report it here?  For instance, if a failed condition is set for more than a brief interval, that might be bad for a conformance test to do.

Reply:
The test at the moment creates the following conditions
```
[]v1.APIServiceCondition{
  v1.APIServiceCondition{Type:"StatusPatched", Status:"True",
    LastTransitionTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC),
    Reason:"E2E", Message:"Set from e2e test"}, 
  v1.APIServiceCondition{Type:"Available", Status:"False",
    LastTransitionTime:time.Date(2022, time.June, 20, 14, 0, 27, 0, time.Local),
    Reason:"ServiceNotFound", Message:"service/e2e-api in \"aggregator-2260\" is not present"}
}
```

Reply:
The `Available: False` is what concerns me - that would trigger alerts / indicate a potential issue, and also possibly impact other tests.  What we need is to either justify that disruption (“it’s ok for this test to create an apiservice that is ‘broken’”) or potentially avoid it (“this test should create an apiservice that is acceptably disabled”).  If the test completes quickly, the former is more acceptable (< 10s).  If the test takes a substantial amount of time, this test is more impactful.  How long does the apiservice created by this test exist for?

Reply:
Ok, if we hit issues we may need to come back and try a different approach, but for now this is better than not testing.

Reply:
@smarterclayton you are right and this test should be tagged as Disruptive, it is breaking the namespace controller deletion, and consequently the CI is having a lot of critical issue s https://github.com/kubernetes/kubernetes/issues/111086#issuecomment-1192166936

Reply:
It’s probably going to be serial, but i’m also not sure why a) the namespace controller failed in this fashion (vs a more graceful one) and b) a service that has never been observed green should show up in discovery for namespace to see.

Reply:
> It’s probably going to be serial, but i’m also not sure why a) the namespace controller failed in this fashion (vs a more graceful one) and b) a service that has never been observed green should show up in discovery for namespace to see.

should we open a new issue to investigate?